### ISSUE TITLE
Very huge issue with the loss function

### ISSUE THREAD

**Issue by kaimatzu** [hongxin001/logitnorm_ood#8]
The way you normalize your logits doesn't result in a valid cumulative distribution function. For this loss function to make sense, you have to apply softmax, log_softmax, or any non-linearity that results in elements of the logits $p$ being $p_i \in 0, R^{+}$, as long as $p \neq \overrightarrow{0}$ (because you can't normalize zero vectors). After all, you do not use negative probabilities in prediction. If the normalized vectors were in the positive orthant of $S^{N-1}$, then it would be a plausible CDF.

**Comment by hongxin001** [comment]
Thank you for the interest. F.crossentropy() has implemented the softmax function in itself, so we don't need to add another softmax function in the code. You may refer to the PyTorch document for details: https://pytorch.org/docs/stable/generated/torch.nn.functional.cross_entropy.html.

**Comment by kaimatzu** [comment]
I see. Forgot about that. Whoops. It's all good then. I'll try it out later on my tasks. I am looking to use it on an experimental network that replaces batchnorm with a instance mean shift + l2 normalization, so it would be fitting if the logits were normalized as well.
